### ISSUE TITLE
Pundit automatically include namespace when self.policy_class is defined in model

### ISSUE THREAD

**Issue by jamesst20** [varvet/pundit#559]
Hi,

I am currently working on a huge Rails project and today we introduced our first STI (Single table inheritance) model.

In other word we have something like this :

```
class MyModelBase < ApplicationRecord; end
class MyModel < MyModelBase; end
class MyModelSomething < MyModelBase; end
```

At first we only had `MyModel < ApplicationRecord` and everything was working well. We had 2 controllers for our `MyModel`. One defined in app/controllers/my_models_controller.rb, one in app/controllers/candidate/my_models_controller.rb. We also had our corresponding policy in app/policies/my_models_policy.rb and app/policies/candidate/my_models_policy.rb.

In our views/layout/application.html.eb we have a `if policy(CipcCheck).reports?` which used to work in every page, but when in MyModelBase we define like so 

```
 def self.policy_class
    CipcCheckPolicy
  end
```

this line  `if policy(CipcCheck).reports?` breaks when we are visiting a page under Canidate::MyModelsController saying `undefined method `reports?' for #<Candidate::CipcCheckPolicy:0x00007fc788b6f9c0>` although we did not request to check the policy under the Candidate namespace.

Removing `self.policy_class` in our MyModelBase fix the issue.

**Comment by Linuus** [comment]
Pundit doesn’t add any namespacing automatically. I think this is just a matter of Ruby’s weird constant lookup.

Can you create a small app which reproduces the issue? It’s a bit hard to follow all the different controllers and models and inheritance in your text :)

**Comment by jamesst20** [comment]
@Linuus 

Thanks for the fast response, I wasn't able to reproduce in an empty project but I could fix it by doing this

```
def self.policy_class
    CipcCheckPolicy
end
```

instead of

```
def self.policy_class
    CipcCheckPolicy
end
```
